Fix `CloudRequestEngine` raising errors outside `Pipeline(...).process()` and stop making requests on its initialization

BohdanVV opened the pull request:
Among the way, method `validate_response` of `CloudRequestEngine` was refactored due to poor quality 

Closes #16

Sniperik committed:
Refactor method `validate_response` of `CloudRequestEngine` and improve code quality

Sniperik committed:
Extend `MockRequestClient` with `server_unavailable` argument to allow testing `CloudRequestEngine` in simulated "server down" mode

Sniperik committed:
Remove unused imports

Sniperik committed:
Convert `flow_element_properties` and `evidence_keys` attributes to the cached properties and add the corresponding test

51Degrees/pipeline-python#16

Sniperik committed:
Add tests to validate functionality of `suppress_process_exceptions` option

51Degrees/pipeline-python#16

Sniperik committed:
Improve code quality/fix linter warnings

Sniperik committed:
Move evidence key filtering from `Evidence.add(...)` to `Evidence.filter()` method

Which is called inside of `FlowData.process()` method, which will resolve any exceptions from `CloudRequestEngine`

51Degrees/pipeline-python#16

justadreamer commented on fiftyone_pipeline_core/fiftyone_pipeline_core/flowdata.py:
```diff
@@ -65,24 +64,25 @@ def process(self):
         if not self.processed:
+            # filter evidence before processing
+            try:
+                self.evidence.filter()
```
please check if there are already tests verifying that evidence filtering works.  if not - please add them.  everything else looks good

BohdanVV replied:
It works, at least these tests are passing

https://github.com/51Degrees/pipeline-python/blob/9fcc7372238e521c77e7f1d7598a34439675498f/fiftyone_pipeline_core/tests/test_core.py#L47-L59

Sniperik committed:
chore: Fix and refactor tests for `fiftyone_pipeline_cloudrequestengine`

Resolves issues caused by changes in `CloudRequestEngine` and addresses poorly written code.

Sniperik committed:
Fix incorrect test of cloud engine

BohdanVV commented:
I've also refactored other tests for `fiftyone_cloudrequestengine`, since they were poorly written

Should be merged now @justadreamer

Sniperik committed:
Move package code to the `src` subfolder, fix coverage report, add default version to `setup.py`

BohdanVV commented:
This PR was extended with the following changes:

- Resolved an issue associated with a deprecated feature in `setuptools >= 69`
- Stripped whitespace characters from the version string coming from `version.txt` to eliminate warnings
- Moved package code to the `src` subfolder to mitigate import conflicts
- Set `pytest` to use the default  `--import-mode=prepend` mode
- Addressed and fixed issues with coverage reports

Sniperik committed:
Change mustache template path in `MANIFEST.in`
